src/helix-ui/elements/HXPopoverElement.js

```diff
@@ -0,0 +1,115 @@
+import { HXElement } from './HXElement';
+import { getPositionWithArrow } from '../util';
+import debounce from 'lodash/debounce';
+import shadowStyles from './_hx-popover.less';
+
+const tagName = 'hx-popover';
+const template = document.createElement('template');
+
+template.innerHTML = `
+  <style>${shadowStyles}</style>
+  <div id="container" class="position-arrow">
+    <slot name="header"></slot>
+    <slot></slot>
+    <slot name="footer"><slot>
+  </div>
+`;
+
+export class HXPopoverElement extends HXElement {
+    static get is () {
+        return tagName;
+    }
+
+    static get observedAttributes () {
+        return [ 'open' ];
+    }
+
+    constructor () {
+        super(tagName, template);
+        this._toggle = this._toggle.bind(this);
+        this._setPosition = this._setPosition.bind(this);
+        this._closeOnBackdropClick = this._closeOnBackdropClick.bind(this);
+    }
+
+    connectedCallback () {
+        this.$defaultAttribute('position', 'bottom-right');
+        this.initialPosition = this.position;
```

Comment:
please prefix `initialPosition` with an underscore to denote that it is an internal/private property